tests/snapshots/compile_file_o_all__exp.hvm.snap

```diff
@@ -2,5 +2,5 @@
 source: tests/golden_tests.rs
 input_file: tests/golden_tests/compile_file_o_all/exp.hvm
 ---
-@main = ({5 ({5 a b} c) (c {5 b d})} (a d))
+@main = ({5 (a {5 b c}) ({5 c d} a)} (d b))
```

Comment:
This should be only `({(a b) a} b)`, aka `@a let {a1 a2} = a; (a1 a2)`

Reply:
I don't understand why it should?
the file is just `main = ((λfλx (f (f x))) (λfλx (f (f x))))`, we don't do anything that could transform to that

Reply:
Sorry, the change above confused me and I thought that was a new file and this was the output